gen_readme.py

```diff
@@ -1,53 +1,46 @@
 #!/bin/python
-import json
-import datetime
-
-grouped_mods = {}
-
-with open("manifest.json", "r", encoding = "UTF-8") as f:
-    MANIFEST = json.load(f)
-    for mod_guid in MANIFEST["mods"]:
-        mod = MANIFEST["mods"][mod_guid]
-        mod["guid"] = mod_guid
 
-        if "flags" not in mod or (
-            "plugin" not in mod["flags"] and
-            "file" not in mod["flags"]
-        ):
-            mods = grouped_mods.get(mod["category"])
-            if mods is None:
-                mods = []
-
-            mods.append(mod)
-            grouped_mods[mod["category"]] = mods
-
-README = None
-with open("gh-pages/.templates/mod-list-template.md", "r", encoding = "UTF-8") as f:
-    README = f.read()
-
-now = datetime.datetime.now(tz=datetime.timezone.utc)
-README += "Last updated at "
-README += f"<time datetime='{now.isoformat()}'>{now.strftime('%d %B %Y, %I:%S')} UTC</time>\n\n"
+"""
+Generates a markdown file to stdout from the JSON manifest passed in with stdin
+"""
 
-for group, mods in grouped_mods.items():
-    mods = mods.sort(key=lambda mod: mod["name"])
+# pylint: disable=redefined-outer-name,broad-except
 
+import json
+import datetime
+import sys
+from typing import Any
+from packaging.version import Version
 
-def should_show_mod(mod):
+def should_show_mod(mod: dict[str, Any]) -> bool:
     """
     Checks if mod should be shown.
 
     Parameters:
     mod: The mod in question
     """
 
+    # Exclude plugins and libraries from being shown
+    if "flags" in mod and (
+        "deprecated" in mod["flags"] or
+        "plugin" in mod["flags"] or
+        "file" in mod["flags"]
+    ):
+        return False
+
+    # Don't add listings for NSFW mods on the website.
+    # NSFW needs to be opt-in to be shown,
+    # mod managers can implement that.
+    if mod["category"] == "NSFW":
+        return False
+
     # Only show mods with versions
     if mod["versions"] is None or len(mod["versions"]) == 0:
         return False
 
     # Don't show mods with only vulnerable versions
     only_vulnerable_versions = True
-    for version in mod["versions"].values():
+    for version in mod["versions"]:
```

Comment:
I think without the `.values()` call this iterates over the keys

Reply:
The mod versions at this point has been turned in a list by `map_mod_versions`, so it iterates over the dict entries in the list.

Reply:
I see that I have been bamboozled